src/base/utils/tar.cpp

```diff
@@ -0,0 +1,230 @@
+        {
+            if (m_header.size() != TAR_HEADER_SIZE)
+                throw std::length_error(std::to_string(m_header.size()));
+
+            // Invalid header: The first header field represents the filename.
+            // If the 1st byte is null then we can't extract a name. So the whole header becomes invalid to us.
+            m_valid = m_header.at(0) != '\0';
+            verifyAndCheckNull();
+        }
+
+        QString name() const
+        {
+            const QByteArray filename = Utils::ByteArray::midView(m_header, TAR_FILENAME_OFFSET, TAR_FILENAME_SIZE);
+            const int filenameLen = customStrlen(filename);
+            if (filenameLen == 0)
+                return {};
+
+            const QByteArray filenamePrefix = Utils::ByteArray::midView(m_header, TAR_FILENAME_PREFIX_OFFSET, TAR_FILENAME_PREFIX_SIZE);
+            const int filenamePrefixLen = customStrlen(filenamePrefix);
+
+            const QString finalFilename = (filenamePrefixLen > 0) ?
+                                        QString::fromUtf8(filenamePrefix.constData(), filenamePrefixLen)
+                                              + '/' + QString::fromUtf8(filename.constData(), filenameLen)
+                                      : QString::fromUtf8(filename.constData(), filenameLen);
+
+            return Utils::Fs::toValidFileSystemName(finalFilename, true);
+        }
+
+        bool isNull() const { return m_null; }
+        bool isValid() const { return m_valid; }
+        bool isVerified() const { return m_verified; }
+        Types type() const { return static_cast<Types>(m_header.at(TAR_FILETYPE_OFFSET)); }
+        int size() const { return octalToDec(Utils::ByteArray::midView(m_header, TAR_FILESIZE_OFFSET, TAR_FILESIZE_SIZE)); }
+
+    private:
+        int octalToDec(const QByteArray &data) const
+        {
+            // only the ASCII characters from '0' to '7' are valid
+            int i = 0;
+            for (; i < data.size(); ++i) { // skip leading invalid bytes
+                char byte = data.at(i);
+                if (byte >= '0' && byte <= '7')
+                    break;
+            }
+
+            int fileSize = 0;
+            for (; i < data.size(); ++i) {
+                char byte = data.at(i);
+                if (byte < '0' || byte > '7') // skip trailing invalid bytes
+                    break;
+
+                fileSize = (8 * fileSize) + (byte - '0');
+            }
+
+            return fileSize;
+        }
+
+        // The various header fields that hold strings are of known max size.
+        // So in the case the string occupies their max size, the string will not
+        // be null terminated. It is pointless to have std::strlen() run through
+        // the whole buffer when we already know the upper limit. Also the buffer
+        // isn't guaranteed to contain a null byte and std::strlen() will have
+        // undefined behavior.
+        int customStrlen(const QByteArray &str) const
```

Comment:
Maybe declare it as `Utils::ByteArray::strlen()`?
Anyway I dislike "custom" prefix here.